common/transport/mqtt/src/test/java/org/thingsboard/server/transport/mqtt/session/GatewaySessionHandlerTest.java

```diff
@@ -15,15 +15,98 @@
  */
 package org.thingsboard.server.transport.mqtt.session;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
 import org.springframework.util.ConcurrentReferenceHashMap;
+import org.thingsboard.server.common.data.id.CustomerId;
+import org.thingsboard.server.common.data.id.DeviceId;
+import org.thingsboard.server.common.data.id.DeviceProfileId;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.transport.TransportService;
+import org.thingsboard.server.common.transport.auth.TransportDeviceInfo;
+import org.thingsboard.server.transport.mqtt.MqttTransportContext;
+
+import java.lang.reflect.Field;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.BDDMockito.willCallRealMethod;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doNothing;
 
 public class GatewaySessionHandlerTest {
 
+    @Mock
+    private TransportService transportService;
+
+    @Mock
+    private DeviceSessionCtx deviceSessionCtx;
+
+    @Mock
+    private MqttTransportContext transportContext;
+
+    private GatewaySessionHandler handler;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+        when(deviceSessionCtx.getSessionId()).thenReturn(UUID.randomUUID());
+        doNothing().when(transportService).recordActivity(any());
+        when(transportContext.getTransportService()).thenReturn(transportService);
+        when(deviceSessionCtx.getContext()).thenReturn(transportContext);
+        handler = new GatewaySessionHandler(deviceSessionCtx, UUID.randomUUID(), true);
+        when(handler.getNodeId()).thenReturn("nodeId");
+    }
+
+    @Test
+    public void shouldRecordActivityWhenOnGatewayPing() throws Exception {
+        // Given
```

Comment:
```
//given when then 
```
Comments are not necessary. You have a well-structured code with straightforward logic.

An additional line break is more than enough to split the logical parts of your test.